### PR TITLE
kubelet/rkt: add support for different stage1 image.

### PR DESCRIPTION
**Pull request by yifan-gu**

### CODE REVIEW

**Comment by jdef** [comment]
not super-familiar with rkt, will re-assign

**Comment by ppalucki** [comment]
@yifan-gu: actually there is some caching issue related to stage1 image naming (look [here](https://github.com/coreos/rkt/issues/1292#issuecomment-132199550) for details) - please take this into consideration when testing

File: pkg/kubelet/rkt/rkt.go [hunk 1]
```diff
@@ -1179,6 +1189,7 @@ func (r *runtime) findRktID(pod *kubecontainer.Pod) (string, error) {
 //  - should we support nsenter + socat in a container, running with elevated privs and --pid=host?
 //
 // TODO(yifan): Merge with the same function in dockertools.
+// TODO(yifan): If the rkt is using lkvm as the stage1 image, then this function will fail.
```
Comment (ppalucki) [inline pkg/kubelet/rkt/rkt.go]: port forwarding is already implemented in lkvm flavor and should work

**Comment by yifan-gu** [comment]
@ppalucki Thanks for heading up. The function breaks mainly because it uses nsenter. But btw I just tried manully, but `rkt status` shows `pid=-1` when run with lkvm stage1. Guess the pid file is not written??

**Comment by yifan-gu** [comment]
@yujuhong Comments addressed.

**Comment by yifan-gu** [comment]
closed in favor of  #13468

**Pull request closed without merging**